msal/src/main/java/com/microsoft/identity/client/GlobalSettings.java

```diff
@@ -0,0 +1,173 @@
+        validateNonNullArgument(context, PublicClientApplication.NONNULL_CONSTANTS.CONTEXT);
+        validateNonNullArgument(listener, PublicClientApplication.NONNULL_CONSTANTS.LISTENER);
+        runOnBackground(new Runnable() {
+            @Override
+            public void run() {
+                synchronized (mGlobalSettingsLock) {
+                    setGlobalConfiguration(
+                            initializeGlobalConfiguration(context, configFileResourceId),
+                            listener
+                    );
+                }
+            }
+        });
+    }
+
+    /**
+     * Load the global configuration file using the configuration file and a listener.
+     *
+     * @param configFile Configuration file.
+     * @param listener Handles success and error messages.
+     */
+    public static void loadGlobalConfigurationFile(@NonNull final File configFile,
+                                                   @NonNull final GlobalSettingsListener listener) {
+        validateNonNullArgument(listener, PublicClientApplication.NONNULL_CONSTANTS.LISTENER);
+        runOnBackground(new Runnable() {
+            @Override
+            public void run() {
+                synchronized (mGlobalSettingsLock) {
+                    setGlobalConfiguration(
+                            initializeGlobalConfiguration(configFile),
+                            listener
+                    );
+                }
+            }
+        });
+    }
+
+    private static void setGlobalConfiguration(@NonNull final GlobalSettingsConfiguration globalConfiguration,
+                                               @NonNull final GlobalSettingsListener listener) {
+        if (mPCAConfigCreated) {
+            listener.onError(new MsalClientException(GLOBAL_INIT_AFTER_PCA,
+                    GLOBAL_INIT_AFTER_PCA_ERROR_MESSAGE));
+        }
+
+        try {
+            validateAccountModeConfiguration(globalConfiguration);
+        } catch (final MsalClientException e) {
+            listener.onError(e);
+            return;
+        }
+
+        mGlobalSettingsConfiguration = globalConfiguration;
+        mGlobalSettingsInitialized = true;
+
+        listener.onSuccess("Global configuration initialized.");
+    }
+
+    private static void runOnBackground(@NonNull final Runnable runnable) {
+        new Thread(runnable).start();
+    }
+
+    private static void validateAccountModeConfiguration(@NonNull final GlobalSettingsConfiguration config) throws MsalClientException {
+        if (config.getAccountMode() == AccountMode.SINGLE
+                && null != config.getDefaultAuthority()
+                && config.getDefaultAuthority() instanceof AzureActiveDirectoryB2CAuthority) {
+            Logger.warn(
+                    TAG,
+                    "Warning! B2C applications should use MultipleAccountPublicClientApplication. "
+                            + "Use of SingleAccount mode with multiple IEF policies is unsupported."
```

Comment:
IEF?

Reply:
I'm not really sure abouyt this one. I copied this from PublicClientApplication. We may not need this check in global? I think it might be getting checked automatically in PCA anyway, after GlobalConfig is merged with PCAConfig.